File: src/1431. Kids With the Greatest Number of Candies/main.py
from typing import List

class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        max_candy = max(candies)

        res: List[bool] = []

        for candy in candies:
            if candy + extraCandies >= max_candy:
                res.append(True)
            else:
                res.append(False)

        return res

File: src/1431. Kids With the Greatest Number of Candies/main.rs
impl Solution {
    pub fn kids_with_candies(candies: Vec<i32>, extra_candies: i32) -> Vec<bool> {
        let max_candy = *candies.iter().max().unwrap();
        candies.into_iter().map(|x| if x + extra_candies >= max_candy { true } else { false }).collect()
    }
}